### PR TITLE
Added `@` import to Webpack config, without needing a package

### PR DESCRIPTION
**Pull request by noviceGuru**
### Added `@` import
It is the default in `NextJS 13` and was not supported by storybook, while I was using it. The way I added it didn't need package to be installed or used

### Corrected the type error in `config.resolve`
Since `config.resolve` can be `undefined`, using `config.resolve.extensions` ran into `TypeScript` error. by adding a condition to the beginning of the line, I got rid of the `TS Error`.

### Worth to mention
I have a greenfield project in hand, using `NextJS13`. After adding `Storybook` for the first components, I ran into an error, that `Storybook` didn't recognize `@` imports that are default in `NextJS13`.
The solution in the documentation didn't work for me, while the one I added worked just fine.

### CODE REVIEW

File: docs/snippets/common/storybook-main-ts-module-resolution.ts.mdx
Comment (jonniebigodes) [inline docs/snippets/common/storybook-main-ts-module-resolution.ts.mdx:13]: @noviceGuru, mind adjusting this snippet's JavaScript version to factor in the same changes here for accuracy and consistency?

File: docs/builders/webpack.md
Comment (jonniebigodes) [inline docs/builders/webpack.md:142]: @noviceGuru we have a bit of room for improvement here, and we can adjust this to the following:
```md
However, if you're working with a framework that provides a default aliasing configuration (e.g., Next.js, Nuxt) and you want to configure Storybook to use the same aliases, you may not need to install any additional packages. Instead, you can extend the default configuration of Storybook to use the same aliases provided by the framework. For example, to set up an alias for the `@` import path, you can add the following to your `.storybook/main.js|ts` file:

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/storybook-main-ts-module-resolution-atsign-import.js.mdx',
    'common/storybook-main-ts-module-resolution-atsign-import.ts.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

```

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@noviceGuru, thanks for taking the time to put together this pull request and help us improve the documentation. Appreciate it 🙏 ! From my end, I left a couple of items that require your attention so that we can merge this.

Additionally, while you're making the changes, do you mind checking the snippets, as they're failing the `pretty-docs` job to address it requires that you run the `yarn pretty-docs` script before committing.

Also, thanks for providing some additional context, and with that, I'm going to loop in a fellow maintainer that worked on the Next.js integration and see if this requires an additional pull request for the Next.Js framework documentation to factor in your issue. Sounds good?

@valentinpalkovic can you chime in on this when you have a moment? Appreciate it.

**Review by jonniebigodes — APPROVED** [review]

File: docs/builders/webpack.md
Comment (jonniebigodes) [inline docs/builders/webpack.md:139]: @noviceGuru can you remove the extra `****` here?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@noviceGuru one small item for you that you left out. And apologize for not mentioning it earlier. Let me know once you've addressed it and I'll merge it.

File: docs/builders/webpack.md
Comment (noviceGuru) [reply docs/builders/webpack.md]: @jonniebigodes It's totally fine, sorry it slipped my hand, lost the notification among the emails.
I corrected it, and had to pull before pushing, so it should merge fast forward. Let me know for any other corrections.

**Review by jonniebigodes — APPROVED** [review]

**Pull request merged**
